Documentation/joss/paper.md

```diff
@@ -19,50 +9,84 @@ authors:
 
-Additionally, OIP incorporates its own EXPRESS parser to automatically generate early-binding code for IFC [Hecht:2019:FBI].
+Let it be stated here that OIP serves as a prototypical playground for developments.
+During the past 3 years, OIP was completely restructured and its code base overhauled - the IFC now stands as the internal model [@Hecht:2019:FBI].
+Major elements were already realigned, while some (previously available) functionality is still considered as work-in-progress.
 
 # Statement of Need
 
-> TODO: A clear Statement of Need that illustrates the research purpose of the software.
+The Architecture, Engineering and Construction (AEC) domain is in its transition from 2D design processes to 3D object-oriented modelling. 
+Building Information Modelling (BIM) is steadily gaining importance replacing the conventional Computer-Aided Design (CAD) practices
+ and getting implemented in every aspect of the very complex software and stakeholder landscape [@Jaud:2019:iccce].
+
+During the development of IFC standard, a clear need for an independent software
+ capable of consuming and producing IFC files according to the newly developed schemata was determined [@Amann:2015:OIP; @Amann:2015:ICCBEI].
+OIP fulfils this role by being open source and based on well known libraries (like Qt and Boost).
 
-[see Valero:2020:ISARC, Hecht:2019:FBI].
+Additionally, the EXPRESS parser provided proved useful for automatically producing
+ source code and data models from newly developed schemata as needed [@vilgertshofer:2017:iwcce].
+
+With the emergence of Scan2BIM and ScanVsBIM research field (deriving from or merely comparing PCD and BIM models),
+ a tool that supports both PCD and IFC data is a welcomed addition to the research processes.
+Recently, OIP was selected among many other software solutions as best suitable for the development of a Scan+BIM platform [@Valero:2020:ISARC].
 
 # Research Projects
 
-> TODO:Mention (if applicable) a representative set of past or ongoing research projects using the software and recent scholarly publications enabled by it.
+The origins of the software date back to the first projects expanding IFC for infrastructure
+ at the Technical University of Munich [@Singer:2014:IfcAlignment; @Amann:2015:OIP; @Amann:2015:OKSTRA; @Amann:2015:ICCBEI].
+There, first implementations of roads' geometric concepts and interfacing between multiple standards (like IFC, OKSTRA and LandXML) have been explored.
+
+Further on, many prototypical additions were implemented during the course of projects:
+
+- support for `IFC4x1`, `IFC4x2`, and `IFC4x3` candidate versions of the IFC standard during their development [@Jaud:2020:siBIM];
+- support for first experimental IFC schema additions for tunnel [@vilgertshofer:2017:iwcce];
+- support for linked data approaches [@Beetz:2019:linkeddata];
+- support for PCD and their analysis [@Hecht:2018:PCdtoAlignment];
+- support for IFC Programming Language (IFC-PL) [@Amann:2018:Diss].
+
```

Comment:
So OKSTRA and LandXML aren't supported anymore?
And thats nothing we could fix in a short amount of time?

Reply:
It's not that simple:

- Both were previously mapped to the internal alignment and terrain models, which have now been deprecated. IFC was mapped as well. These were basically very simple common-denominator-models (which was enough for the prototyping).
- We've switched to IFC being the internal model, so the mapping needs updating. This is a bit more difficult since previously, a lot of content from all the standards was ignored during the mapping. Now, this would have to be done properly, filling in all corresponding data in IFC with defaults (if not present in OKSTRA), etc.